src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleCreationExpression.cs

```diff
@@ -13,8 +9,80 @@ internal partial class LocalRewriter
     {
         public override BoundNode VisitTupleCreationExpression(BoundTupleCreationExpression node)
         {
-            var rewrittenArguments = VisitList(node.Arguments);
-            return new BoundObjectCreationExpression(node.Syntax, node.ConstructorOpt, rewrittenArguments);
+            ImmutableArray<BoundExpression> rewrittenArguments = VisitList(node.Arguments);
+            return RewriteTupleCreationExpression(node, rewrittenArguments);
+        }
+
+        /// <summary>
+        /// Converts the expression for creating a tuple instance into an expression creating a ValueTuple (if short) or nested ValueTuples (if longer).
+        ///
+        /// For instance, for a long tuple we'll generate:
+        /// creationExpression(ctor=largestCtor, args=firstArgs+(nested creationExpression for remainder, with smaller ctor and next few args))
+        /// </summary>
+        private BoundNode RewriteTupleCreationExpression(BoundTupleCreationExpression node, ImmutableArray<BoundExpression> rewrittenArguments)
+        {
+            int remainder;
+            int chainLength = TupleTypeSymbol.NumberOfValueTuples(rewrittenArguments.Length, out remainder);
```

Comment:
It feels that the call to NumberOfValueTuples is unnecessary. The chainLength is amount of types in the underlyingTupleTypeChain builder after GetUnderlyingTypeChain call. remainder is the Arity of the type we are about to construct.